extension/test/data_test.ts

```diff
@@ -85,6 +85,67 @@ describe('data.ts', function () {
     });
   });
 
+  describe('parseWordDictEntry', function () {
+    describe('with a valid word dict line', function () {
+      it('parses a word dict entry from a word dict line which has a word, pronunciation, and definitions', function () {
+        const wordDictLine =
+          '<word> [<pronunciation>] /<definition-1>/<definition-2>/<definition-3>/';
+
+        const wordDictEntry = parseWordDictEntry(wordDictLine);
+
+        expect(wordDictEntry).to.deep.equal({
+          word: '<word>',
+          pronunciation: '<pronunciation>',
+          definitions: ['<definition-1>', '<definition-2>', '<definition-3>'],
+        });
+      });
+
+      it('parses a word dict entry from a word dict line which has a word and definitions, but no pronunciation', function () {
+        const lineFromWordDictWithNoPronunciation =
+          '<word> /<definition-1>/<definition-2>/';
+
+        const wordDictEntry = parseWordDictEntry(
+          lineFromWordDictWithNoPronunciation
+        );
+
+        expect(wordDictEntry).to.deep.equal({
+          word: '<word>',
+          pronunciation: null,
+          definitions: ['<definition-1>', '<definition-2>'],
+        });
+      });
+    });
+
+    describe('with an invalid word dict line', function () {
+      it('returns null when you pass in an empty word dict line', function () {
+        const emptyWordDictLine = '';
+
+        const wordDictEntry = parseWordDictEntry(emptyWordDictLine);
+
+        expect(wordDictEntry).to.equal(null);
+      });
+
+      it('returns null when you pass in a word dict line with badly formatted definitions', function () {
```

Comment:
I wonder if it's worth being explicit about 'badly formatted definitions'.  I guess it's having a separator but empty defintions? If it's hard to think of a good descriptor this isn't bad but I thought of it while reading.

Reply:
I reworded this. Thanks!